### PR TITLE
OETHb: Tooling and contracts setup

### PR DESCRIPTION
**Pull request by shahthepro**
## Changes overview
- Rename contracts with `Base` in them to `Abstract`
- Carry forward wOETH deployment artifacts  from https://github.com/OriginProtocol/origin-dollar/pull/2022
- Move base deployments to its own directory
- Add fork test fixtures and CI setup
- Add deploy tooling for Base
- Add OETHBaseOracleRouter (similar to `OETHOracleRouter`)
- Add deployment files for Vault, OETHb, wOETHb and OracleRouter

## Code Change Checklist

To be completed before internal review begins:

- [ ]  The contract code is complete
- [ ]  Executable deployment file
- [ ]  Fork tests that test after the deployment file runs
- [ ]  Unit tests *if needed
- [ ]  The owner has done a [full checklist review](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) of the code + tests

Internal review:

- [ ] Two approvals by internal reviewers

## Deploy checklist

Two reviewers complete the following checklist:

```
- [ ] All deployed contracts are listed in the deploy PR's description
- [ ] Deployed contract's verified code (and all dependencies) match the code in master
- [ ] The transactions that interacted with the newly deployed contract match the deploy script.
- [ ] Governance proposal matches the deploy script
- [ ] Smoke tests pass after fork test execution of the governance proposal
```

### CODE REVIEW

File: contracts/utils/deploy-l2.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+const { isFork, isArbFork, isBaseFork } = require("../test/helpers");
```
Comment (sparrowDom) [inline contracts/utils/deploy-l2.js:1]: very cool this is a separate file

**Review by sparrowDom — APPROVED** [review]
great job LGTM

File: contracts/contracts/oracle/OETHBaseOracleRouter.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     * @return maxStaleness maximum acceptable data staleness duration
+     */
+    function feedMetadata(address asset)
+        internal
+        view
+        virtual
+        override
+        returns (address feedAddress, uint256 maxStaleness)
+    {
+        if (asset == 0x4200000000000000000000000000000000000006) {
```
Comment (clement-ux) [inline contracts/contracts/oracle/OETHBaseOracleRouter.sol:62]: Maybe we can avoid the "magical address" and create a variable for it?

File: contracts/contracts/oracle/OETHBaseOracleRouter.sol [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            // FIXED_PRICE: WETH/ETH
+            feedAddress = FIXED_PRICE;
+            maxStaleness = 0;
+        } else if (asset == 0xD8724322f44E5c58D7A815F542036fb17DbbF839) {
```
Comment (clement-ux) [inline contracts/contracts/oracle/OETHBaseOracleRouter.sol:66]: Same here for magical address.

File: contracts/contracts/oracle/OETHBaseOracleRouter.sol [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            // Chainlink: https://data.chain.link/feeds/base/base/woeth-oeth-exchange-rate
+            // Bridged wOETH/OETH
+            feedAddress = 0xe96EB1EDa83d18cbac224233319FA5071464e1b9;
```
Comment (clement-ux) [inline contracts/contracts/oracle/OETHBaseOracleRouter.sol:69]: Same here for magical address.

**Review by clement-ux — COMMENTED** [review]
Looks good to me, just a comment about "magical address".

File: contracts/contracts/oracle/OETHBaseOracleRouter.sol (hunk 2 above)
Comment (shahthepro) [reply contracts/contracts/oracle/OETHBaseOracleRouter.sol:62]: Good point, I can move it to a constant

**Pull request merged**